### ISSUE TITLE
Grid-container; use a wrapper-div or the body-element?

### ISSUE THREAD

**Issue by senolv** [rachelandrew/cssgrid-ama#148]
Hi Rachel, I read a lot of these articles [MDN - CSS Grid Layout](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout).

I want to know why you all use a wrapper-div for the container, is this only for demonstrating purposes or has it an important reason?

Thanks

**Comment by xlf1024** [comment]
The wrapper can be anything, including the `<body>` element. Considering that `<div>`s carry no semantics and only the styling of `display:block`, which is overridden anyways, there is probably no important reason except for making the examples generic.

In prod, lay out your html according to semantics first. Then use whatever happens to be the parent of your grid items as the container, for main page layout this might well be the `<body>`, but you can use grid anywhere, e.g. on a `<nav>`, an `<article>` or anything else. Common are also `<ol>` and `<ul>` as containers with `<li>`s as items, this would e.g. be appropriate for an article overview. There, you'd have to set `list-style: none` to remove the markers.
However, you might need to add wrapper-divs if the parent of your grid-items has also non-grid-items, though I can't think of a use case for this, or you somehow need additional boxes.
Benefits of this approach are a) screenreader support, b) better performance in no-css scenarios (e.g. Reader mode, outdated browsers or command-line and c) possibly SEO, as semantic html helps the search engine understand your page better.

For examples, look at e.g. [labs.jensimmons.org](labs.jensimmons.org). The experiment lists in each section are `<ul>`s with `display:grid`. The header has an additional wrapper, which was needed for the rotation, without it, the header's background would have been rotated as well.

Given that Rachel and Jen appear to be friends I guess her approach would be similar to Jen's.

Tl;dr: `<div>`s and `<span>`s are the most generic elements in html, if it works with them, it works with basically anything else.

P.S.: in contrast to a `<div>`, the `<body>` is full-width by default. You can change that though.

**Comment by rachelandrew** [comment]
> Given that Rachel and Jen appear to be friends I guess her approach would be similar to Jen's.

is a very strange assertion. Please don't decide for me what my approach to anything might be, thanks!
